packages/apps/job-launcher/server/src/modules/payment/payment.service.ts

```diff
@@ -253,7 +252,7 @@ export class PaymentService {
     });
 
     const totalAmount = paymentEntities.reduce((total, payment) => {
-      return add(total, mul(payment.amount, payment.rate));
+      return add(total, payment.amount);
```

Comment:
In getUserBalance we were multiplying the amount in dollar with the token rate, so we were mixing dollar amounts with token amounts.
- Be sure to record payment amounts in the currency of the transaction without additional multiplication by the exchange rate.
- This is incorrect logic, since we need to know the resulting balance of the user, given the fact that we are working with unchanged transaction data.

Reply:
There something confusing here:

- As far as I understood amount field in payments table always stores USD values, since the balance we show to users is USD this would make sense to me. Might have missunderstood this.
- If I was wrong with 1., there is something wrong in create fortune job endpoint. The fundAmount value should be USD, and following your logic we store it as hmt (look screenshots taken with the code before this pr, payment id=4)

![image](https://github.com/humanprotocol/human-protocol/assets/61605646/70ff7ec2-49c6-4941-b7a9-d76e130da23e)
![image](https://github.com/humanprotocol/human-protocol/assets/61605646/fd49d2a2-dc80-4a8a-9bf3-01725fd859ac)

Reply:
Agree with screenshot, because this topic related with user balance. If the user what to add any fiat to his account he can do it: usd, eur and etc. And we need to record it as original data with original topup currency and rate relative to usd.
The same for crypto deposit. 
Since we are converting all amounts to usd and showing to the user as a balance. If the user want to deposit escrow, we are withdrawing an amount in the currency in which the escrow was issued, in our case it is hmt, and we store these values ​​in the database in relation to the currency used by the user, in our case it is usd.

Reply:
let's make this logic clear.

Reply:
the amounts are saved in the original currency in which the user made, for this reason we save the rate.